go/logic/inspect.go

```diff
@@ -521,17 +522,28 @@ func (this *Inspector) estimateTableRowsViaExplain() error {
 }
 
 // CountTableRows counts exact number of rows on the original table
-func (this *Inspector) CountTableRows() error {
+func (this *Inspector) CountTableRows(ctx context.Context) error {
 	atomic.StoreInt64(&this.migrationContext.CountingRowsFlag, 1)
 	defer atomic.StoreInt64(&this.migrationContext.CountingRowsFlag, 0)
 
 	log.Infof("As instructed, I'm issuing a SELECT COUNT(*) on the table. This may take a while")
 
 	query := fmt.Sprintf(`select /* gh-ost */ count(*) as rows from %s.%s`, sql.EscapeName(this.migrationContext.DatabaseName), sql.EscapeName(this.migrationContext.OriginalTableName))
 	var rowsEstimate int64
-	if err := this.db.QueryRow(query).Scan(&rowsEstimate); err != nil {
+	if err := this.db.QueryRowContext(ctx, query).Scan(&rowsEstimate); err != nil {
```

Comment:
To the best of my understanding, the `go-sql-driver` does not actually kill the query when context is cancelled, it is on the user to issue a `KILL` query. In which case we should first obtain a `Connection` so that we can get its connection_id, which we can then use to kill it.

Reply:
You're right! I regret this statement 😅 :

> Update on the above: I smartened up and read the docs instead of the code, which say go-sql-driver does support cancelling queries via contexts

I think the row count has to be pushed into a goroutine then, so `CountTableRows` can simultaneously wait on it and on a signal to kill the count query.